packages/devtools_app/lib/src/profiler/flutter/cpu_profiler.dart

```diff
@@ -22,42 +23,90 @@ class CpuProfiler extends StatelessWidget {
 
   final Function(CpuStackFrame stackFrame) onStackFrameSelected;
 
+  static const Key expandButton = Key('CpuProfiler - Expand Button');
+  static const Key collapseButton = Key('CpuProfiler - Collapse Button');
+
   // TODO(kenz): the summary tab should be available for UI events in the
   // timeline.
-  static const cpuProfilerTabs = [
+  static const tabs = [
     Tab(text: 'CPU Flame Chart'),
     Tab(text: 'Call Tree'),
     Tab(text: 'Bottom Up'),
   ];
 
   static const emptyCpuProfile = 'No CPU profile data';
 
+  @override
+  _CpuProfilerState createState() => _CpuProfilerState();
+}
+
+class _CpuProfilerState extends State<CpuProfiler>
+    with SingleTickerProviderStateMixin {
+  TabController _tabController;
+
+  @override
+  void initState() {
+    super.initState();
+    _tabController = TabController(length: CpuProfiler.tabs.length, vsync: this)
+      ..addListener(() {
+        setState(() {});
+      });
+  }
+
+  @override
+  void dispose() {
+    super.dispose();
+    _tabController.dispose();
+  }
+
   @override
   Widget build(BuildContext context) {
     final textTheme = Theme.of(context).textTheme;
-    return DefaultTabController(
-      length: cpuProfilerTabs.length,
-      child: Column(
-        crossAxisAlignment: CrossAxisAlignment.start,
-        children: <Widget>[
-          TabBar(
-            labelColor: Theme.of(context).textTheme.body1.color,
-            isScrollable: true,
-            tabs: cpuProfilerTabs,
-          ),
-          Expanded(
-            child: _buildCpuProfileDataView(textTheme),
-          ),
-        ],
-      ),
+    return Column(
+      crossAxisAlignment: CrossAxisAlignment.start,
+      children: <Widget>[
+        Row(
+          mainAxisAlignment: MainAxisAlignment.spaceBetween,
+          children: [
+            TabBar(
+              labelColor: Theme.of(context).textTheme.body1.color,
+              isScrollable: true,
+              controller: _tabController,
+              tabs: CpuProfiler.tabs,
+            ),
+            if (_tabController.index != 0)
```

Comment:
The order of the tabs is different on the timeline and the performance page. This should instead check that the selected tab is not the flame chart tab.

Reply:
Done, using a key.

Note that the current implementation is using the same tab list on both pages.  If we want a different order, we'll need to provide an alternate tabs list.